### ISSUE TITLE
Hexchat is not starting truly minimized when set to do so.

### ISSUE THREAD

**Issue by NolanSyKinsley** [hexchat/hexchat#1451]
I know this may be a minor annoyance or something considered close to a gripe, but when I have hexchat set to start minimized to the taskbar under linux when my system boots, hexchat opens a window visibly then minimizes the window to the task bar.

I know this may not seem like much, but hexchat is a fairly popular program, and when we want to show off our impressive boot speeds under linux the large flash of the hexchat windows seems like a hiccough that should not happen, and feels jarring and unsightly to an otherwise fast and smooth boot process.

If I have hexchat set to start minimized then no window should be created at all until I request it.

**Comment by TingPing** [comment]
Looks like this was initially done for the minimize to tray option since the tray is implemented in a plugin (it waits until the window is created to load plugins so they have a valid window I assume). There is technical reason that the other minimize level has to wait for this that I can think of.

**Comment by abhigenie92** [comment]
Has this been fixed?

**Comment by TingPing** [comment]
@abhigenie92 Is the issue open?

**Comment by NolanSyKinsley** [comment]
yes, this issue is still open and has not been fixed. If my hunch is correct it is with network joining. It seems that the window will stay open until a placeholder has been spawned for each network joined. It doesn't require an actual connection, but the tab/placeholder must be spawned before the window minimizes.

**Comment by deekej** [comment]
Hello,

this is also a major annoyance for me on Windows (8.1/10). When this happens, hexchat freezes on that window, and it can take up to ~5 seconds before it disappears... :-/

Could we please get a fix for this?

**Comment by NolanSyKinsley** [comment]
Something changed in a recent update. The window still appears, but it vanishes very quickly, where before it would hang around for several seconds.

**Comment by TingPing** [comment]
@NolanSyKinsley Just a guess but #1819 probably made that a lot faster.

**Comment by NolanSyKinsley** [comment]
I was mistaken. It was something with my system. For about 5 boots it was really fast, just a flash. Now it has returned to normal as it was before.

**Comment by SadaleNet** [comment]
This issue is very annoying to me.

I use an IRC bouncer for offline messages notification. I'm expecting to be notified whenever someone highlight me while I'm offline, or whenever someone PM me while I'm offline. However, Hexchat receives the offline messages while its window is opened before it finishes joining the channels and minimized.

Since the messages were received while the window is opened, no system notification popup were displayed. All I get is a beep for every offline message. And I couldn't figure out which channel is source of the message unless it check the channels one by one. :(

I'm considering switching to another IRC client because of this limitation. :(

**Comment by SadaleNet** [comment]
I'm sorry for being a bit rude. It turned out that it takes more work for me to switch client than to fix the bug. 😛

Pull request sent. I've briefly tested it. It seems to work fine. Anyway, it's always a good idea to further test it. I'm sorry that I do not have the time to do so. 😞

**Comment by NolanSyKinsley** [comment]
Brilliant! Just compiled and tested on Manjaro, works perfectly! Looking over the code it all looks fine, nothing out of line. Absolutely should be pushed to main in my opinion. Thank you @SadaleNet!

**Comment by deekej** [comment]
*Sigh* I've just switched to different app recently... :( Sorry.

**Comment by SadaleNet** [comment]
@deekej You switched because of this bug?

**Comment by deekej** [comment]
@SadaleNet Yes, I did. I was using the hexchat on Windows at my home, so when moving to Windows 10 I decided to use Pidgin instead for my IRC chats... :-/ (This bug was really annoying for me after every login, because it was stealing focus from other apps.)

**Comment by mohammadg** [comment]
This should be fixed by now, its really a basic but important bug to fix.

**Comment by SadaleNet** [comment]
@mohammadg not really. It will take a while until it get merged due to technical issues. My modification to the code base isn't good enough to meet the quality standard of the code of Hexchat.

It will probably take me around a week until I have free time to improve my modification.
